CASSGO-76: Add Makefile

dkropachev opened the pull request:
Implement all CICD steps in Makefile and make workflows use Makefile. 
It will help with local development and reproducing CICD issues locally.

Ticket: https://issues.apache.org/jira/browse/CASSGO-76

dkropachev commented:
@tengu-alt, can you please review it

joao-r-reis commented:
This seems like a reasonable proposal but I'd like to see some kind of discussion in a C* dev mailing list thread or ASF `#cassandra-drivers` slack channel to see if there is broad community support for this

dkropachev commented:
> This seems like a reasonable proposal but I'd like to see some kind of discussion in a C* dev mailing list thread or ASF `#cassandra-drivers` slack channel to see if there is broad community support for this

Can you invite me to the workspace and channel, you can find my email on the account.

dkropachev commented:
@joao-r-reis , can you please invite me to the slack, so that I could start discussion there ?

joao-r-reis commented:
There's a [link on the repo's README file](https://github.com/apache/cassandra-gocql-driver/blob/trunk/README.md?plain=1#L4)

dkropachev commented:
> There's a [link on the repo's README file](https://github.com/apache/cassandra-gocql-driver/blob/trunk/README.md?plain=1#L4)

@joao-r-reis , that link is not working:
```
This invite has expired
We let the person who sent this invite know that you tried to join, so you may hear from them again soon.
```

joao-r-reis commented:
I'm getting an error saying that a user with that email address is already on the slack workspace, you should be able to log in with that email address I think?

dkropachev commented:
> I'm getting an error saying that a user with that email address is already on the slack workspace, you should be able to log in with that email address I think?

Thanks someone already invited me

jameshartig commented on conn_test.go:
Are these changes necessary?

dkropachev replied:
otherwise CICD would fail

joao-r-reis replied:
Why? CICD isn't failing currently on trunk, is it on a specific go version?

dkropachev replied:
There is no such step on CICD on trunk

joao-r-reis replied:
Can you clarify? Are you saying this test isn't running in CICD on trunk atm?

dkropachev replied:
Sorry for missleading, in current cicd it is run without tags https://github.com/apache/cassandra-gocql-driver/blob/05e6036ff1d7d3d8aaacbe6817723e435cad22db/.github/workflows/main.yml#L25-L25

while in this PR it runs with all known tags:
```
	go vet --tags "unit all cassandra integration" ./...
```

Since this file is tagged https://github.com/apache/cassandra-gocql-driver/blob/05e6036ff1d7d3d8aaacbe6817723e435cad22db/conn_test.go#L1-L2
It is not looked at by `go vet` in current CICD

joao-r-reis replied:
Oh that's a good catch

dkropachev committed:
Move all cicd steps into Makefile

1. Implement all steps in Makefile
2. Make CICD use Makefile
3. Drop setup-environment step, it is done automatically now
4. Add linting step

dkropachev committed:
conn_test: address linter complaints
